homeassistant/components/comelit/coordinator.py

```diff
@@ -71,16 +71,14 @@ def platform_device_info(
     async def _async_update_data(self) -> dict[str, Any]:
         """Update router data."""
         _LOGGER.debug("Polling Comelit Serial Bridge host: %s", self._host)
+        logged = False
         try:
             logged = await self.api.login()
         except (asyncio.exceptions.TimeoutError, aiohttp.ClientConnectorError) as err:
             _LOGGER.warning("Connection error for %s", self._host)
             raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
+        finally:
```

Comment:
Does this also run when an exception is raised? Also The data update coordinator already catches TimeoutError and ClientConnectorError

Reply:
Yes it does and is neded for example for a wrong password. Its not an exception but sets the logged bool to false.